src/ratelimit/RateLimitController.ts

```diff
@@ -0,0 +1,186 @@
+import type { Patch } from 'immer';
+
+import { BaseController } from '../BaseControllerV2';
+
+import type { RestrictedControllerMessenger } from '../ControllerMessenger';
+
+/**
+ * @type RateLimitState
+ * @property requests - Object containing number of requests in a given interval for each origin and api type combination
+ */
+export type RateLimitState<T extends string> = {
+  requests: Record<T, Record<string, number>>;
+};
+
+export interface CallArgs<T> {
```

Comment:
We've been trying to use descriptive names for generics, so we should not use `T` for generics in this type or elsewhere in this file. In general, I believe that single-character generic parameter names are only appropriate for certain, simple utility types.

(I used name my generics in this manner too because it appears to a common practice, but when you stop to think about it, why would generics be an exception to the general rule of using descriptive names? It's bizarre!)

Reply:
Done!